Source/WebGPU/WGSL/WGSL.cpp

```diff
@@ -54,19 +55,19 @@ SuccessfulCheck::SuccessfulCheck(Vector<Warning>&& messages, UniqueRef<AST::Shad
 
 SuccessfulCheck::~SuccessfulCheck() = default;
 
-PrepareResult prepare(const AST::ShaderModule& ast, const HashMap<String, PipelineLayout>& pipelineLayouts)
```

Comment:
I think we should try to keep these const. One of the lessons I've learned from the LFC work we've been doing in WebKit is that it's really important to be able to have strict contracts about what is populated when and by which thing; and the C++ type system is a great way of enforcing it. Conceptually, the AST is the input to the preparation function, and conceptually, preparation shouldn't modify the AST; if we want to fill in some previously-unpopulated fields inside the AST, I think we should use other side data structures for those. Having strong contracts around this stuff makes the code way more easy to hack on; you don't have to manually figure out which field was written by which stage.